### DISCUSSION TITLE
HDFS related questions

### DISCUSSION THREAD

**Discussion by zhougit86** [risingwavelabs/risingwave#10566] [Q&A]
We are trying to introduce risingwave into our production enviroment. But before start, we want to understand 2 things

1, can we use hdfs as the statebackend?
2, can we dump result to aliyun maxcompute?

**Comment by lmatz** [comment]
> can we use hdfs as the statebackend?

Yes, Risingwave suppors HDFS as the state backend, please check more details at:
https://www.risingwave.dev/docs/current/risingwave-kubernetes/#deploy-a-risingwave-instance

**Comment by lmatz** [comment]
> can we dump result to aliyun maxcompute?

Do you mean `Write data to OSS` more precisely?

If you are referring to `create sink XXX` and sink output to `OSS`, Risingwave right now does not support that.
Please correct me if I understand you wrong, thanks!

**Reply by zhougit86** [reply]
yes, I mean the max computer sink. Does risingwave support custom sink like flink do? 

 The first step we might plan to build some etl job, they extract from kafka to aliyun odps. As these simple tasks can give us some confidence. The user defined sink might provide a very good start to us.

**Reply by StrikeW** [reply]
Right now we don't support sink to odps, and it is not trivial to add one. Maybe you can extract from kafka and sink to a kafka topic then ingest to odps via other tools. I think aliyun should support ingest to odps via kafka.

**Reply by lmatz** [reply]
If the target destination, e.g. `OSS`, is S3-compatible, we can consider supporting an S3 sink to avoid the need of writing a customized sink, just an early thought.

Wonder if this `user-defined sink` has any other requirements, or could you elaborate on more details of the `dump result` part?

**Reply by zhougit86** [reply]
https://nightlies.apache.org/flink/flink-docs-release-1.17/docs/dev/table/sourcessinks/ 

this kind of source/sink is good enough

**Comment by zhougit86** [comment]
Furthermore, I read 
'RisingWave offers not only streaming ETL and streaming analytics, but also online serving capabilities through its built-in batch query engine.'    in https://www.risingwave.dev/docs/current/risingwave-flink-comparison/#typical-use-cases

Does this mean, risingwave could provide same ETL function with less performance cost than flink does?

**Reply by lmatz** [reply]
> risingwave could provide same ETL function with less performance cost than flink does?

Yes, Risingwave is written in Rust(versus Java) and has a native storage engine that is optimized for stream processing.

>  but also online serving capabilities

By saying so, we want to emphasize that the results produced by Flink can only be queried when the results are output and stored into another storage system, e.g. Redis or Cassandra or any other DBs.

While in Risingwave, the results produced by Risingwave can be directly stored in Materialized Views. So you can directly query against Materialized Views to see your latest result with a consistency guarantee. No third-party system needed.

**Reply by zhougit86** [reply]
Understand, I have 2 more questions:
1. can we support java udf
2. can we specify psql address in play mode? as I want to run RW in a linux env and use psql in a mac env

**Reply by zhougit86** [reply]
Can some master provide some of the IM tools? like feishu or dingding, I might need some help in my POC process

**Reply by StrikeW** [reply]
Hi @zhougit86 , I sent a email to you, please check your mail.

**Reply by zhougit86** [reply]
Ok, I have contact you according to your suggestion.

**Comment by wcy-fdu** [comment]
Glad to hear that you plan to try RisingWave, may I ask about your HDFS configuration? For example, whether there is namenode HA, etc.

**Reply by zhougit86** [reply]
I will have further discussion later on this topic, As this is more "product" related case.
